### PR TITLE
Correctly format files and ranges with line endings other than LF

### PR DESCRIPTION
**Pull request by ccordoba12**
- `black.format_file_contents` (the function we're using to format files and ranges) fails for CRLF and CR. So we need to convert line endings to LF before using it and restore them afterwards.
- I noticed this while I was investigating the bug reported in spyder-ide/spyder#17143.
- Drop support for Python 3.6 because this change requires python-lsp-server 1.4.0+, which also dropped support for that version.

### CODE REVIEW

File: pylsp_black/_utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Eol chars accepted by the LSP protocol
+EOL_CHARS = ["\r\n", "\r", "\n"]
+
+
+def get_eol_chars(text):
```
Comment (ccordoba12) [inline pylsp_black/_utils.py:5]: I also need to add this function in `python-lsp-server` to check for eols there.

Should we import it from there or do you prefer to leave it here?

Comment (ccordoba12) [reply pylsp_black/_utils.py:5]: See python-lsp/python-lsp-server#151

**Comment by haplo** [comment]
Thank you for the PR @ccordoba12, I will test it out as soon as I can, but I might not be able to do it until the weekend.

My initial thought is that I would expect black to handle EOL itself. But maybe that only works when black opens the file itself, but we are passing raw input text instead. I would like to understand the internals better.

I would also like to run some microbenchmarks for this code to make it as fast as possible, as it will be called all the time. The [timeit module](https://docs.python.org/3/library/timeit.html?highlight=timeit#module-timeit) might be enough.

File: pylsp_black/_utils.py (hunk 1 above)
Comment (ccordoba12) [reply pylsp_black/_utils.py:5]: @krassowski already did a lot of benchmarks and the regex one is way more efficient. So, I'll use his version and import it from `python-lsp-server` here.

**Comment by ccordoba12** [comment]
Moving to 1.2.0 due the urgency of releasing 1.1.0 given #29.

File: pylsp_black/_utils.py (hunk 1 above)
Comment (haplo) [reply pylsp_black/_utils.py:5]: Great, thanks for the update.

**Comment by ccordoba12** [comment]
@haplo, this is ready for review.

File: pylsp_black/plugin.py [hunk 2]
```diff
@@ -6,9 +6,11 @@
 import black
 import toml
 from pylsp import hookimpl
+from pylsp._utils import get_eol_chars
```
Comment (haplo) [inline pylsp_black/plugin.py:9]: I with the module was `pylsp.utils` and not `_utils`, but not much we can do about that at this point.

**Review by haplo — APPROVED** [review]
LGTM :raised_hands:

File: pylsp_black/plugin.py (hunk 2 above)
Comment (ccordoba12) [reply pylsp_black/plugin.py:9]: You're right. We inherited that from the old Palantir project and now it's used everywhere in our fork.

**Comment by ccordoba12** [comment]
Thanks @haplo for the review!

Next week I'll work on the branch you created to add config options, so we can release 1.2.0 afterwards.

**Pull request merged**